src/plugins/add-cal/add-calendar.js

```diff
@@ -0,0 +1,155 @@
+ */
+( function( $, wb ) {
+"use strict";
+
+/*
+ * Variable and function definitions.
+ * These are global to the plugin - meaning that they will be initialized once per page,
+ * not once per instance of plugin on the page. So, this is a good place to define
+ * variables that are common to all instances of the plugin on a page.
+ */
+var componentName = "wb-addcal",
+	selector = ".provisional." + componentName,
+	initEvent = "wb-init." + componentName,
+	$document = wb.doc,
+
+	/**
+	* @method init
+	* @param {jQuery Event} event Event that triggered the function call
+	*/
+	init = function( event ) {
+
+		// Start initialization
+		// returns DOM object = proceed with init
+		// returns undefined = do not proceed with init (e.g., already initialized)
+		var elm = wb.init( event, componentName, selector ),
+			$elm = $( elm );
+
+		if ( elm ) {
+
+			wb.ready( $( elm ), componentName );
+
+			var properties = elm.querySelectorAll( "[property]" ),
+				event_details = new Object(),
+				i,
+				i_len,
+				prop_cache,
+				googleLink,
+				icsFile,
+				i18nDict = {
+					en: {
+						"addcal-addto": "Add to",
+						"addcal-calendar": "calendar",
+						"addcal-ical": "iCal format (iPhone, Outlook...)"
+					},
+					fr: {
+						"addcal-addto": "Ajouter au",
+						"addcal-calendar": "calendrier",
+						"addcal-ical": "Format iCal (iPhone, Outlook....)"
+					}
+				};
+
+			// Initiate dictionary
+			i18nDict = i18nDict[ $( "html" ).attr( "lang" ) || "en" ];
+			i18nDict = {
+				addto: i18nDict[ "addcal-addto" ],
+				calendar: i18nDict[ "addcal-calendar" ],
+				ical: i18nDict[ "addcal-ical" ]
+			};
+
+			//Set date stamp with the date modified
+			if ( $( "time[property='dateModified'][datetime]" ).length ) {
+				event_details.dtStamp = new Date( $( "time[property='dateModified']" ).attr( "datetime" ) );
+			} else {
+				event_details.dtStamp = new Date( $( "time[property~='dateModified']" ).text() );
+			}
+
+			event_details.dtStamp = event_details.dtStamp.toISOString().replace( /\..*[0-9]/g, "" ).replace( /-|:|\./g, "" );
+
+			i_len = properties.length;
+			for ( i = 0; i < i_len; i++ ) {
+				prop_cache = properties[ i ];
+				switch ( prop_cache.getAttribute( "property" ) ) {
+				case "name":
+					if ( $( prop_cache ).parentsUntil( ( "." + componentName ), "[typeof=Place]" ).length ) {
+						event_details.place = prop_cache.textContent;
+					} else {
+						event_details.name = prop_cache.textContent;
+					}
+					break;
+				case "description":
+					event_details.description = prop_cache.textContent.replace( /(\r\n|\n|\r)/gm, " " );
+					break;
+				case "startDate":
+					event_details.sDate = new Date( prop_cache.getAttribute( "datetime" ) ).toISOString().replace( /\..*[0-9]/g, "" ).replace( /-|:|\./g, "" );
+					break;
+				case "endDate":
+					event_details.eDate = new Date( prop_cache.getAttribute( "datetime" ) ).toISOString().replace( /\..*[0-9]/g, "" ).replace( /-|:|\./g, "" );
+					break;
```

Comment:
May be it will worth to create a function to parse the date, you seems to apply the same date + replace operation 3 times